### PR TITLE
Add a test for thinlto + gold + buildid

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: tests/buildid.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Checks that we generate BuildID
+//
+// REQUIRES: clang, lld
+// RUN: %clang -fuse-ld=lld -flto -O2 %S/Inputs/foo.c %s -o %t
+// RUN: %t
+// RUN: file %t|grep -q BuildID
```
Comment (serge-sans-paille) [inline tests/buildid.c:6]: The output of `file` is not reliable across systems (on my system, your test fails). What's the goal of this test? Check that a GNU_BUILD_ID note is set? If so, maybe rely on the output of `readelf -a` (or better, of `llvm-objdump`) ?

Comment (sylvestre) [reply tests/buildid.c:6]: Correct, I want to test buildid, it caused me some pain when I moved to stage2  (I still have nightmare ;)

Comment (sylvestre) [reply tests/buildid.c]: fixed

File: tests/gold.c [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Test gold support
+// REQUIRES: clang
+// RUN: %clang -fuse-ld=gold %s -o %t
+// RUN: %t
+
+#include <stdio.h>
+int main() {
+if (1==1) {
```
Comment (serge-sans-paille) [inline tests/gold.c:8]: I wonder what you're trying to test with that trivial comparison?

Comment (sylvestre) [reply tests/gold.c:8]: I just took the code coverage test. The code is unrelevant here :)

File: tests/gold.c [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Test gold support
```
Comment (serge-sans-paille) [inline tests/gold.c:1]: Some extra thoughts: we should check (at configure time) that the gold linker is available, and `REQUIRE` ir.

**Pull request merged**